torch_xla/distributed/xla_backend.py

```diff
@@ -82,10 +82,8 @@ def allgather(self, output_tensors_list, input_tensors):
   # https://github.com/pytorch/pytorch/blob/release/1.10/torch/distributed/distributed_c10d.py#L1129
   def broadcast(self, tensors, opts):
     root_tensor = tensors[opts.rootTensor]
-    root_rank = opts.rootRank
-    if root_rank != self.rank():
-      with torch.no_grad():
-        root_tensor.zero_()
+    scale = 1 if opts.rootRank != self.rank() else 0
+    root_tensor = root_tensor * scale
```

Comment:
Should we have torch.no_grad() here to omit these steps from backward?

Reply:
Done